toolchain/lower/file_context.cpp

```diff
@@ -118,6 +127,14 @@ auto FileContext::Run() -> std::unique_ptr<llvm::Module> {
                               /*Priority=*/0);
   }
 
+  if (cpp_code_generator_) {
+    cpp_code_generator_->HandleTranslationUnit(cpp_ast()->getASTContext());
+    CARBON_CHECK(!llvm::Linker::linkModules(
+        /*Dest=*/*llvm_module_,
+        /*Src=*/std::unique_ptr<llvm::Module>(
+            cpp_code_generator_->ReleaseModule())));
```

Comment:
(Also here: please don't put code with side effects inside a CHECK.)

Reply:
Thanks, done.
Is this because the side effect might run more than once on failure?